### PR TITLE
Change webiste title dynamically

### LINKED ISSUE #495: 🐛 [Bug report]: Website title is the same on every page
**Describe the bug**
The `title` for the website on every page is the same _(ReactPlay - One app to learn, create, and share ReactJS projects)_. 

**To Reproduce**
1. Go to any page on the website, apart from the homepage.
2. The title is same for every page

**Expected behavior**
The title should change according the page that the user is currently in.
**For example:** 
If a user visits the play lists page, the title should display something like "Explore Plays - ReactPlay". Or if a user visits a play page, it should display the name of the page as the title.

### PR DESCRIPTION
**Pull request by ammaaraslam**
# Description

Added functionality to dynamically change the `title` of the website according to the page. If the user is on the `ideas` page, then the `title` of the page is "ReactPlay - Ideas"

Fixes #495 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tested Locally.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by ammaaraslam** [comment]
I found that in production, the title changes on the play page but not on other pages. Please wait till I find a fix..Thanks

**Comment by ammaaraslam** [comment]
> I found that in production, the title changes on the play page but not on other pages. Please wait till I find a fix..Thanks

Fixed

**Comment by ammaaraslam** [comment]
@atapas and @koustov, the issue has been fixed and this PR is now ready for review.

**Comment by atapas** [comment]
@ammaaraslam 

The build fails with this error



Creating an optimized production build...
--
14:13:31.965 | Failed to compile.
14:13:31.966 |  
14:13:31.966 | Module not found: Error: Package path ./dist/loader/css/react-spinner-loader.css is not exported from package /vercel/path0/node_modules/react-loader-spinner (see exports field in /vercel/path0/node_modules/react-loader-spinner/package.json)

**Comment by ammaaraslam** [comment]
> @ammaaraslam
> 
> The build fails with this error
> 
> ## Creating an optimized production build...
> 14:13:31.965 | Failed to compile. 14:13:31.966 |   14:13:31.966 | Module not found: Error: Package path ./dist/loader/css/react-spinner-loader.css is not exported from package /vercel/path0/node_modules/react-loader-spinner (see exports field in /vercel/path0/node_modules/react-loader-spinner/package.json)

That's weird, it got published yesterday, let me check.

**Comment by ammaaraslam** [comment]
Hi @atapas. I don't seem to figure out why the build is failing. The build is failing after you have merged from the `main` branch. Perhaps a change from that is what causing the build to fail. Could you check on that and let me know.

**Comment by atapas** [comment]
> Hi @atapas. I don't seem to figure out why the build is failing. The build is failing after you have merged from the `main` branch. Perhaps a change from that is what causing the build to fail. Could you check on that and let me know.

Yeah I see in general all build fails today due to the same error. Now sure if there is a new version of react-loader-spinner library that is broken. Do you want to check that out locally with a yarn build?

**Comment by atapas** [comment]
> > Hi @atapas. I don't seem to figure out why the build is failing. The build is failing after you have merged from the `main` branch. Perhaps a change from that is what causing the build to fail. Could you check on that and let me know.
> 
> Yeah I see in general all build fails today due to the same error. Now sure if there is a new version of react-loader-spinner library that is broken. Do you want to check that out locally with a yarn build?

As suspected it is an issue with the library. They have published a new version and it is breaking things. I see an open issue already. Until they fix the open issue, please revert the version back to what was working and fix to it.

https://github.com/mhnpd/react-loader-spinner/issues/135

**Comment by ammaaraslam** [comment]
@atapas, I have reverted the version of `react-loader-spinner` and now the build is a success.

**Comment by atapas** [comment]
> Visit Preview

Yes, I see that.. let's create an issue and fix this one alone so that we can merge it to unblock other branches.

**Comment by ammaaraslam** [comment]
> > Visit Preview
> 
> Yes, I see that.. let's create an issue and fix this one alone so that we can merge it to unblock other branches.

@atapas, I have created and issue and PR to fix this issue.

File: src/meta/DefMeta.jsx [hunk 1]
```diff
@@ -1,8 +1,33 @@
 import { Helmet } from "react-helmet";
+import { useState, useEffect } from "react";
+import { useLocation } from "react-router-dom";
 
 function DefMeta() {
+  const [title, setTitle] = useState("ReactPlay - One app to learn, create, and share ReactJS projects")
+  const currentPath = useLocation().pathname
+
+  useEffect(() => {
+    if (currentPath === "/plays") {
```
Comment (atapas) [inline src/meta/DefMeta.jsx:10]: It works but I think we can simplify the code a bit here.

How about a simple JSON map with key is the current path and and value the title. We can have it managed outside of this file altogether... so passing the current path we get the title, and if it is undefined, we set the default title.

This way, tomorrow a new path comes, this file no need to go through changes for it. What say?

Comment (ammaaraslam) [reply src/meta/DefMeta.jsx:10]: That's a great idea, didn't think of that. I'll make the changes and update you.

Comment (ammaaraslam) [reply src/meta/DefMeta.jsx:10]: @atapas, for this simplification, I'm thinking of an array that holds the path and it's corresponding title. Then we check if the current path is in the this array and then assign the relevant title, if the path is not defined then we set a default title. So if we add a new page in the future, all we have to do is add the path and title for it inside the array. Is this what you have suggested? Let me know your final thoughts.

**Comment by ammaaraslam** [comment]
@atapas and @koustov. This PR is now ready for review, I have made changes to simplify the logic as Tapas suggested

**Comment by ammaaraslam** [comment]
@atapas and @koustov. This PR is now ready for review, I have made changes to simplify the logic as Tapas suggested

**Review by atapas — APPROVED** [review]
LGTM

**Pull request merged**